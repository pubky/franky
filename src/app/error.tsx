'use client';

import { AppError, CommonErrorType } from '@/lib/error';
import { useEffect } from 'react';

export default function Error({ error, reset }: { error: Error; reset: () => void }) {
  useEffect(() => {
    // Convert to AppError if it's not already one
    if (!(error instanceof AppError)) {
      new AppError(CommonErrorType.UNEXPECTED_ERROR, error.message || 'An unexpected error occurred', 500, {
        originalError: error,
      });
    }
  }, [error]);

  return (
    <div className="flex min-h-screen items-center justify-center bg-background p-5">
      <div className="w-full max-w-2xl rounded-lg bg-card p-8 text-center shadow-md">
        <h1 className="mb-4 text-xl font-semibold text-destructive">Something went wrong</h1>
        <p className="mb-6 text-muted-foreground">
          {error instanceof AppError ? error.message : 'An unexpected error occurred'}
        </p>
        {process.env.NODE_ENV === 'development' && (
          <details className="mb-6 text-left">
            <summary className="cursor-pointer text-primary hover:underline">Error Details</summary>
            <pre className="mt-2 overflow-x-auto rounded-md bg-muted p-4 text-sm">
              {error instanceof AppError ? JSON.stringify(error, null, 2) : error.toString()}
            </pre>
          </details>
        )}
        <button
          onClick={reset}
          className="rounded-md bg-primary px-4 py-2 text-primary-foreground transition-colors hover:bg-primary/90"
        >
          Try Again
        </button>
      </div>
    </div>
  );
}
