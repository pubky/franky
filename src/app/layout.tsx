import { Inter_Tight } from 'next/font/google';
import { DatabaseProvider } from '@/providers';

import './globals.css';

import * as Organisms from '@/organisms';
import * as Molecules from '@/molecules';
import * as Atoms from '@/atoms';

const interTight = Inter_Tight({
  variable: '--font-geist-sans',
  subsets: ['latin'],
});

export default function RootLayout({ children }: { children: React.ReactNode }) {
  return (
    <RootContainer>
      <DatabaseProvider>
        <Molecules.Metadata />
        <Organisms.Header />
        {children}
        <Molecules.Toaster />
      </DatabaseProvider>
    </RootContainer>
  );
}

export function RootContainer({ children }: { children: React.ReactNode }) {
  return (
    <Atoms.Container as="html" lang="en" className="dark">
      <Atoms.Container as="body" className={`${interTight.variable} antialiased`}>
        <Molecules.PageContainer>{children}</Molecules.PageContainer>
      </Atoms.Container>
    </Atoms.Container>
  );
}
