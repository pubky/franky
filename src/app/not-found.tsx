import * as Molecules from '@/molecules';

export const metadata = Molecules.Metadata({
  title: 'Page Not Found - Pubky App',
  description: 'The page you are looking for does not exist.',
  robots: false,
});

export default function NotFound() {
  return (
    <div className="flex h-screen flex-col items-center justify-center">
      <h2 className="text-2xl font-bold">Not Found</h2>
      <p>Could not find requested resource</p>
    </div>
  );
}
