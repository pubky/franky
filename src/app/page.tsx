'use client';

import { Header, HeroSection } from '@/components/ui';

export default function Home() {
  const handleSignIn = () => {
    console.log('Sign in clicked');
  };

  const handleCreateAccount = () => {
    console.log('Create account clicked');
  };

  return (
    <>
      {/* Full viewport background */}
      <div
        className={`opacity-10 lg:opacity-100 fixed inset-0 bg-cover bg-center bg-no-repeat -z-10`}
        style={{
          backgroundImage: 'url(/images/bg-home.svg)',
        }}
      />

      <div className="dark min-h-screen bg-transparent z-10">
        <Header onSignIn={handleSignIn} githubUrl="#" twitterUrl="#" telegramUrl="#" />

        <main className="relative">
          <HeroSection
            title={
              <>
                <span className="text-brand">Unlock</span>
                <br />
                the web.
              </>
            }
            subtitle="Pubky requires an invite code"
            onSignIn={handleSignIn}
            onCreateAccount={handleCreateAccount}
          />
        </main>
      </div>
    </>
  );
}
