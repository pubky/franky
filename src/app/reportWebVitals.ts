import type { NextWebVitalsMetric } from 'next/app';

import { recordTiming } from '../../benchmarks/reporters/jsonReporter';

const HYDRATION_METRIC_NAMES = new Set(['Next.js-hydration', 'Hydration']);

export function reportWebVitals(metric: NextWebVitalsMetric) {
  recordTiming('web-vitals', metric.name, metric.value, {
    id: metric.id,
    label: metric.label,
  });

  if (HYDRATION_METRIC_NAMES.has(metric.name)) {
    recordTiming('hydration', metric.name, metric.value, {
      id: metric.id,
      label: metric.label,
    });
  }
}
