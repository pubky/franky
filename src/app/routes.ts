export const ROOT_ROUTES = '/';

export enum ONBOARDING_ROUTES {
  BACKUP = '/onboarding/backup',
  HOMESERVER = '/onboarding/homeserver',
  INSTALL = '/onboarding/install',
  PROFILE = '/onboarding/profile',
  PUBKY = '/onboarding/pubky',
  SCAN = '/onboarding/scan',
}

export enum AUTH_ROUTES {
  SIGN_IN = '/sign-in',
  LOGOUT = '/logout',
}

export enum APP_ROUTES {
  HOME = '/home',
  SEARCH = '/search',
  HOT = '/hot',
  BOOKMARKS = '/bookmarks',
  SETTINGS = '/settings',
  PROFILE = '/profile',
  PROFILE_EDIT = '/profile/edit',
  PROFILE_POSTS = '/profile/posts',
  PROFILE_REPLIES = '/profile/replies',
  PROFILE_NOTIFICATIONS = '/profile/notifications',
  PROFILE_FOLLOWERS = '/profile/followers',
  PROFILE_FOLLOWING = '/profile/following',
  PROFILE_FRIENDS = '/profile/friends',
  PROFILE_TAGGED = '/profile/tagged',
}

export enum POST_ROUTES {
  POST = '/post',
}

export const PUBLIC_ROUTES: string[] = [AUTH_ROUTES.LOGOUT];

export const ALLOWED_ROUTES = [
  ONBOARDING_ROUTES.PROFILE,
  APP_ROUTES.HOME,
  APP_ROUTES.SEARCH,
  APP_ROUTES.HOT,
  APP_ROUTES.BOOKMARKS,
  APP_ROUTES.SETTINGS,
  APP_ROUTES.PROFILE,
  APP_ROUTES.PROFILE_EDIT,
  APP_ROUTES.PROFILE_POSTS,
  APP_ROUTES.PROFILE_REPLIES,
  APP_ROUTES.PROFILE_NOTIFICATIONS,
  APP_ROUTES.PROFILE_FOLLOWERS,
  APP_ROUTES.PROFILE_FOLLOWING,
  APP_ROUTES.PROFILE_FRIENDS,
  APP_ROUTES.PROFILE_TAGGED,
  POST_ROUTES.POST,
  AUTH_ROUTES.LOGOUT,
];

// Route guard configurations for different authentication states
export const UNAUTHENTICATED_ROUTES = {
  allowedRoutes: [
    ROOT_ROUTES,
    AUTH_ROUTES.SIGN_IN,
    ONBOARDING_ROUTES.INSTALL,
    ONBOARDING_ROUTES.SCAN,
    ONBOARDING_ROUTES.PUBKY,
    ONBOARDING_ROUTES.BACKUP,
    ONBOARDING_ROUTES.HOMESERVER,
    AUTH_ROUTES.LOGOUT,
  ],
  redirectTo: ROOT_ROUTES,
};

export const AUTHENTICATED_ROUTES = {
  allowedRoutes: ALLOWED_ROUTES,
  redirectTo: APP_ROUTES.HOME,
};

// Backwards compatibility
export const HOME_ROUTES = {
  HOME: APP_ROUTES.HOME,
};
