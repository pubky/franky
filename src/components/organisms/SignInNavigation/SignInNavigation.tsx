'use client';

import { useRouter } from 'next/navigation';

import * as Atoms from '@/atoms';
import * as Organisms from '@/organisms';
import * as App from '@/app';

export const SignInNavigation = () => {
  const router = useRouter();

  const handleRestore = () => {
    router.push(App.HOME_ROUTES.HOME);
  };

  return (
    <Atoms.Container className="flex-col-reverse md:flex-row gap-3 lg:gap-6 justify-start">
      <Atoms.Container className="flex-col gap-3 sm:flex-row w-auto sm:w-full justify-start items-start mx-0 sm:mx-auto">
        <Organisms.DialogRestoreRecoveryPhrase onRestore={handleRestore} />
        <Organisms.DialogRestoreEncryptedFile onRestore={handleRestore} />
      </Atoms.Container>
    </Atoms.Container>
  );
};
