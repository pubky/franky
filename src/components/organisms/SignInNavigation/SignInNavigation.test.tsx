import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi, beforeEach } from 'vitest';
import { SignInNavigation } from './SignInNavigation';
import * as App from '@/app';

// Mock Next.js router
const mockPush = vi.fn();
vi.mock('next/navigation', () => ({
  useRouter: () => ({
    push: mockPush,
  }),
}));

// Minimal atoms
vi.mock('@/atoms', () => ({
  Container: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="container" data-class-name={className}>
      {children}
    </div>
  ),
}));

// Use real libs, only stub cn for deterministic classes if used by atoms callers
vi.mock('@/libs', async (importOriginal) => {
  const actual = await importOriginal<typeof import('@/libs')>();
  return {
    ...actual,
    cn: (...classes: (string | undefined)[]) => classes.filter(Boolean).join(' '),
  };
});

// Stub child dialogs so we can trigger onRestore
vi.mock('@/organisms', () => ({
  DialogRestoreRecoveryPhrase: ({ onRestore }: { onRestore?: () => void }) => (
    <button data-testid="restore-phrase" onClick={onRestore}>
      Restore Phrase
    </button>
  ),
  DialogRestoreEncryptedFile: ({ onRestore }: { onRestore?: () => void }) => (
    <button data-testid="restore-file" onClick={onRestore}>
      Restore File
    </button>
  ),
}));

describe('SignInNavigation', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('renders both restore dialogs', () => {
    render(<SignInNavigation />);

    expect(screen.getByTestId('restore-phrase')).toBeInTheDocument();
    expect(screen.getByTestId('restore-file')).toBeInTheDocument();
  });

  it('pushes to HOME on restore (phrase)', () => {
    render(<SignInNavigation />);

    fireEvent.click(screen.getByTestId('restore-phrase'));
    expect(mockPush).toHaveBeenCalledWith(App.HOME_ROUTES.HOME);
  });

  it('pushes to HOME on restore (file)', () => {
    render(<SignInNavigation />);

    fireEvent.click(screen.getByTestId('restore-file'));
    expect(mockPush).toHaveBeenCalledWith(App.HOME_ROUTES.HOME);
  });
});

describe('SignInNavigation - Snapshots', () => {
  it('matches snapshot', () => {
    const { container } = render(<SignInNavigation />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
