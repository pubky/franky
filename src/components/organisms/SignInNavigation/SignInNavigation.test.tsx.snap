// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignInNavigation - Snapshots > matches snapshot 1`] = `
<div
  data-class-name="flex-col-reverse md:flex-row gap-3 lg:gap-6 justify-start"
  data-testid="container"
>
  <div
    data-class-name="flex-col gap-3 sm:flex-row w-auto sm:w-full justify-start items-start mx-0 sm:mx-auto"
    data-testid="container"
  >
    <button
      data-testid="restore-phrase"
    >
      Restore Phrase
    </button>
    <button
      data-testid="restore-file"
    >
      Restore File
    </button>
  </div>
</div>
`;
